src/sources/http_server.rs

```diff
@@ -283,32 +294,136 @@ impl SourceConfig for SimpleHttpConfig {
     }
 }
 
-fn add_path(events: &mut [Event], key: &str, path: &str) {
-    for event in events.iter_mut() {
-        event
-            .as_mut_log()
-            .try_insert(key, Value::from(path.to_string()));
-    }
+#[derive(Clone)]
+struct SimpleHttpSource {
+    headers: Vec<String>,
+    query_parameters: Vec<String>,
+    path_key: String,
+    decoder: Decoder,
+    log_namespace: LogNamespace,
 }
 
-fn add_headers(events: &mut [Event], headers_config: &[String], headers: HeaderMap) {
-    for header_name in headers_config {
-        let value = headers.get(header_name).map(HeaderValue::as_bytes);
+impl SimpleHttpSource {
+    /// Enriches the passed in events with metadata for the `request_path` and for each of the headers.
+    fn enrich_events(&self, events: &mut [Event], request_path: &str, headers: HeaderMap) {
+        for event in events.iter_mut() {
+            let log = event.as_mut_log();
+
+            // add request_path to each event
+            self.log_namespace.insert_source_metadata(
+                SimpleHttpConfig::NAME,
+                log,
+                Some(LegacyKey::InsertIfEmpty(path!(self.path_key.as_str()))),
+                path!("path"),
+                request_path.to_owned(),
+            );
+
+            // add each header to each event
+            for header_name in &self.headers {
+                let value = headers.get(header_name).map(HeaderValue::as_bytes);
+
+                self.log_namespace.insert_source_metadata(
+                    SimpleHttpConfig::NAME,
+                    log,
+                    Some(LegacyKey::InsertIfEmpty(path!(header_name))),
```

Comment:
Since we have a chance to update schemas in the vector namespace, would it make sense to nest the header names under a top level key like "headers"? It might be rare, but we could theoretically run into collisions here.

Reply:
For both of these we are adding a new collection to scope them?

Reply:
I'm on the same page as Spencer here- the schema and in the inserting of fields, we are scoping that to a collection of "headers" and "query_parameters" respectively.
Are you seeing something we aren't? :thinking: 

> It might be rare, but we could theoretically run into collisions here.

Good observation. We aren't checking for any duplicated headers/query_parameters that the user would have to have configured.
I guess we could add that? Kind of tangential to this work but, I'm thinking it could just be a simple `warn!` log to inform the user that they specified the same key more than once.

Reply:
This is inserting all the headers into `headers` whereas I presume it should insert into `headers.<header_name>`.

Reply:
Doh, you right @StephenWakely it should be something like: https://github.com/vectordotdev/vector/blob/master/src/sources/aws_s3/sqs.rs#L552 for both headers and query params

Reply:
>   This is inserting all the headers into headers whereas I presume it should insert into headers.<header_name>.

Good catch!